### DISCUSSION TITLE
call predict function

### DISCUSSION THREAD

**Discussion by kghezelbash** [hezarai/hezar#175] [Q&A] [answered]
Hi, is there any way to call the predict function on Image to Text (OCR) model with the image object and not its address?

**Comment by arxyzan** [comment]
Hi @kghezelbash,
Yes, you can pass in the numpy image array or PIL image objects instead of address. It also supports passing lists of image arrays or PIL images.

**Comment by kghezelbash** [comment]
I have written this code:
```
from hezar.models import Model
# OCR with CRNN
model = Model.load("hezarai/crnn-fa-printed-96-long")
image = cv2.imread("/content/drive/MyDrive/id_data_ocr_section/06872c53fb2cbb613f0652e6f798995c9b73e62ffea593edb681860109571f81.jpeg")
texts = model.predict(image[780:870, 550:850])
print(texts)
```
and the output is :

[{'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}, {'text': ''}]

but when I save the image and pass the address it reads the image fine, I cant find my mistake.

**Comment by arxyzan** [comment, marked as answer]
@kghezelbash, If you want to pass in a numpy array image, you have to pass it inside a list:
```python
from hezar.models import Model
# OCR with CRNN
model = Model.load("hezarai/crnn-fa-printed-96-long")
image = cv2.imread("/content/drive/MyDrive/id_data_ocr_section/06872c53fb2cbb613f0652e6f798995c9b73e62ffea593edb681860109571f81.jpeg")
texts = model.predict([image[780:870, 550:850]])
print(texts)
```

**Comment by kghezelbash** [comment]
Thank you =))))))
